include/wil/registry.h

```diff
@@ -17,16 +17,2967 @@
+         * \param subKey The name of a subkey that this function opens or creates
+         *        Note: this cannot be null (see the above referenced API documentation)
+         * \param[out] hkey A reference to a wil::unique_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT create_unique_key_nothrow(HKEY key, PCWSTR subKey, ::wil::unique_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(subKey, hkey.put(), access);
+        }
+
+#if defined(__WIL_WINREG_STL)
+        /**
+         * \brief Opens a new HKEY to the specified path - see RegOpenKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subKey The name of the registry subkey to be opened
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::shared_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT open_shared_key_nothrow(HKEY key, _In_opt_ PCWSTR subKey, ::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.open_key(subKey, hkey.put(), access);
+        }
+
+        /**
+         * \brief Creates a new HKEY to the specified path - see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subKey The name of a subkey that this function opens or creates
+         *        Note: this cannot be null (see the above referenced API documentation)
+         * \param[out] hkey A reference to a wil::shared_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT create_shared_key_nothrow(HKEY key, PCWSTR subKey, ::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(subKey, hkey.put(), access);
+        }
+#endif // #define __WIL_WINREG_STL
+
+        namespace reg_query_info_key_args
+        {
+            // RegQueryInfoKeyW has a long list of optional parameters
+            static constexpr PWSTR null_class{ nullptr };

```

Comment:
use `inline` here. `static` variables produce a unique copy in every translation unit, binary bloat.

Reply:
ok. I removed named variables here.